mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -0,0 +1,225 @@
+        return call_endpoint(get_databricks_host_creds(),
+                             endpoint, method, json_body, response_proto)
+
+    def _get_write_credentials(self, run_id, path=None):
+        json_body = message_to_json(GetCredentialsForWrite(run_id=run_id, path=path))
+        return self._call_endpoint(DatabricksMlflowArtifactsService,
+                                   GetCredentialsForWrite, json_body)
+
+    def _get_read_credentials(self, run_id, path=None):
+        json_body = message_to_json(GetCredentialsForRead(run_id=run_id, path=path))
+        return self._call_endpoint(DatabricksMlflowArtifactsService,
+                                   GetCredentialsForRead, json_body)
+
+    def _extract_headers_from_credentials(self, credential):
+        headers = dict()
+        for header in credential.headers:
+            headers[header.name] = header.value
+        return headers
+
+    def _azure_upload_file(self, credentials, local_file, artifact_path):
+        """
+        Uploads a file to a given Azure storage location.
+
+        The function uses a file chunking generator with 100 MB being the size limit for each chunk.
+        This limit is imposed by the stage_block API in azure-storage-blob.
+        In the case the file size is large and the upload takes longer than the validity of the
+        given credentials, a new set of credentials are generated and the operation continues. This
+        is the reason for the first nested try-except block
+
+        Finally, since the prevailing credentials could expire in the time between the last
+        stage_block and the commit, a second try-except block refreshes credentials if needed.
+        """
+        try:
+            headers = self._extract_headers_from_credentials(credentials)
+            service = BlobClient.from_blob_url(blob_url=credentials.signed_uri, credential=None,
+                                               headers=headers)
+            uploading_block_list = list()
+            for chunk in yield_file_in_chunks(local_file, _AZURE_MAX_BLOCK_CHUNK_SIZE):
+                block_id = base64.b64encode(uuid.uuid4().hex.encode())
+                try:
+                    service.stage_block(block_id, chunk, headers=headers)
+                except ClientAuthenticationError:
+                    _logger.warning(
+                        "Failed to authorize request, possibly due to credential expiration."
+                        "Refreshing credentials and trying again..")
+                    credentials = self._get_write_credentials(self.run_id,
+                                                              artifact_path).credentials.signed_uri
+                    service = BlobClient.from_blob_url(blob_url=credentials, credential=None)
+                    service.stage_block(block_id, chunk, headers=headers)
+                uploading_block_list.append(block_id)
+            try:
+                service.commit_block_list(uploading_block_list)
```

Comment:
After doing some digging through the docs, I think we should include the headers on the `commit_block_list` operation, since these headers are likely to be used for encryption information; I'd imagine that this information is just as relevant for a block commit as it is for a block upload. Sorry for the runaround here.

Reply:
Done.